### ISSUE TITLE
full card reload any time an entity has a value change

### ISSUE THREAD

**Issue by Koendejongh** [iantrich/config-template-card#78]
Hi, it seems since HA version 2021.9.0 the config-template-card fully reloads as soon as any entity has a value change. It worked perfect in previous versions. 

I use this card in my floorplan and uses the light entities to update the floorplan. 

Since this version, it gives the blue round circles when in loading phase (it did not in previous HA version) and it will reload (disappear and start over again building the card) as soon as it gets triggered by an entity.

Thanks in advance for support.

**Comment by MaxxKra** [comment]
I have the same problem since updating to 2021.9.x.
The UI works without "config-template-card", but I need templates to display the LED colors 
![Dashboard_good](https://user-images.githubusercontent.com/83531755/132184883-9bc8517b-e9a9-4a13-b707-79c4acd1d0eb.PNG)
![Dashboard_broken](https://user-images.githubusercontent.com/83531755/132184901-e77c4484-0097-4242-8849-8b754dd26ae1.PNG)

**Comment by iantrich** [comment]
@bramkragten you happen to know what might have caused this off the top of your head? I haven't been keeping up on frontend changes as of late. Thanks!

**Comment by iantrich** [comment]
Possible Lovelace issue: https://github.com/home-assistant/frontend/issues/9988

**Comment by bkr1969** [comment]
I have no problem with the entities initially loading, but I too have the entire screen reloaded each time a state changes or is even polled. It's particularly bad when my roomba is running and the screen reloads literally every second. I skipped core 2021.8 due to thermostat issues but I did not have this issue on 2021.7

**Comment by bkr1969** [comment]
@bramkragten has closed all issues with this relating to front end.  Definitely needs to be looked at from the card side, likely combined with front end.

**Comment by smonesi** [comment]
I forced `home-assistant-frontend==20210818.0` (but kept everything else aligned to 2021.9) and the issue disappeared. 

It is still present with `home-assistant-frontend==20210825.0`, so the issue is caused by something committed between those two tags (I see that https://github.com/home-assistant/frontend/pull/9690 contains a change regarding ha-circular-progress but I have no real clue if this is related or not).

**Comment by bkr1969** [comment]
Can you please tell me how you did the 0818.0 frontend? I skipped all 08 due to issues with my thermostat integration.  Those were fixed with 09 but if I'm able to just revert the frontend and keep the thermostats working, that would be great until this issue is fixed.

**Comment by Koendejongh** [comment]
Yes, please let me know how to force to a previous version.

**Comment by Koendejongh** [comment]
Also I reprogrammed my floorplan without config-template-card, but just the picture elements card and I do not get the spinning circles and it doesn't reload the card anymore. 
But it also does not update when a light is being turned on. And of course the "mix-blend-mode: lighten" function doesn't work. That's why I need the config-template-card.

**Comment by smonesi** [comment]
> Yes, please let me know how to force to a previous version.

I have a manual installation of Home Assistant in a python virtual environment. In the root directory of the python env, I edited `lib64/python3.9/site-packages/homeassistant/package_constraints.txt` replacing `home-assistant-frontend==20210830.0` with `home-assistant-frontend==20210818.0` and I also changed in the same way the `lib64/python3.9/site-packages/homeassistant/components/frontend/manifest.json` file.

Then I run `pip3 install home-assistant-frontend==20210818.0` although I think this step is useless (it was the first thing I tried but without changing the files above, the latest version was automatically installed).

I'm not sure this is the best way to do this (and I'm not even sure if it may cause unwanted side-effects), but for now it looks like a good workaround.

**Comment by bkr1969** [comment]
Unfortunately that will not work for me.  :-|

**Comment by iantrich** [comment]
Could someone try to get me a really simple config that replicates this issue? Thanks.

**Comment by bkr1969** [comment]
About as simple as I can get.

```
type: custom:config-template-card
entities:
  - switch.ceiling_fan
card:
  type: picture-elements
  elements:
    - type: image
      tap_action:
        action: toggle
      entity: switch.ceiling_fan
      image: /local/floorplan/famfanoff.png
      state_image:
        'on': /local/floorplan/famfanon.png
      style:
        top: 80.3%
        left: 77.6%
  image: https://demo.home-assistant.io/stub_config/floorplan.png
```
I included my animated fan.png

![famfanon](https://user-images.githubusercontent.com/42385989/132936335-60faa3b0-6b3a-4e70-8e21-f907a3973a9f.png)

**Comment by Koendejongh** [comment]
Hi, if you take the example of Brian and replace the fan switch with a light switch you can physical control on the wall, you will see it will reload the floorplan each time you switch the light.

**Comment by Nantero1** [comment]
+1 on this one, same problems here.

**Comment by lockyt** [comment]
Getting the same issues.  I will also add that it flattens the battery on my iPad in the wall faster than it will charge when I have HA app loaded and displayed.  Doesn't when I close it down - so lots of processing going on.

**Comment by MaxxKra** [comment]
With version 2021.9.6, the error no longer occurred with my configuration. I don't know what was changed but it seems to work.

**Comment by Koendejongh** [comment]
I am also running 2021.9.6 and the reloading is still there, although it seems to be a bit quicker.  I can still trigger the reloading by switching on a light.

**Comment by Nantero1** [comment]
> With version 2021.9.6, the error no longer occurred with my configuration. I don't know what was changed but it seems to work.

Nope. Issue is still there. Of course the minimal configuration is minimal, but with a more sophisticated configuration the reloading is a big pain.

**Comment by TCoughlin02** [comment]
I am having the same issue, even after updating to 2021.9.6

**Comment by Graeme-SStrong** [comment]
Same issue for me after updating to 2121.9.6. Rolled back to 2021.8.8 and the problem has gone.

**Comment by smonesi** [comment]
I had a deeper look at https://github.com/home-assistant/frontend/pull/9690 and although I'm not really a Java/TypeScript developer I believe this PR does more than "Only update cameras when they are visible" as it introduces the concept of image "loading" (and the spinning image) for all the images, not just cameras.

It looks like the "loading" code is triggered when a picture-elements is first loaded but, for some reason, also every time it is re-loaded due to its content being changed by the surrounding config-template-card. This affects all the images content of the picture-elements, including the main image whose height is also reset to the default 100px during load (hence the "squeeze" effect shown in the screenshot above).

Unfortunately I am not able to even try to fix either the hui-image code (by removing the "loading" feature on static images, that seems a bit unnecessary) or the config-template-card (if possible, to avoid it triggering a complete reload of its inner card).

**Comment by iantrich** [comment]
I'll try to take a look tonight or tomorrow.

**Comment by Koendejongh** [comment]
> I had a deeper look at https://github.com/home-assistant/frontend/pull/9690 and although I'm not really a Java/TypeScript developer I believe this PR does more than "Only update cameras when they are visible" as it introduces the concept of image "loading" (and the spinning image) for all the images, not just cameras.
> 
> It looks like the "loading" code is triggered when a picture-elements is first loaded but, for some reason, also every time it is re-loaded due to its content being changed by the surrounding config-template-card. This affects all the images content of the picture-elements, including the main image whose height is also reset to the default 100px during load (hence the "squeeze" effect shown in the screenshot above).
> 
> Unfortunately I am not able to even try to fix either the hui-image code (by removing the "loading" feature on static images, that seems a bit unnecessary) or the config-template-card (if possible, to avoid it triggering a complete reload of its inner card).
> 

Good work. You might involve @bramkragten if that didn't happen yet.

**Comment by iantrich** [comment]
Took a quick glance at my code and I believe the issue is that the card is re-rending on updates. Not sure why I designed it that way, but that's super inefficient. It's gone unnoticed because it was rendered so quickly, I'm guessing. A fix should be easy enough for the performance aspect. For those that have images not loading at all and stay as loading symbols, might something more to that I think.

**Comment by bkr1969** [comment]
@iantrich, thanks for your time on this.  I have found that most people who are involved with HA are pretty committed to it despite being a non-paying job.  Looking forward to this fix!

**Comment by iantrich** [comment]
Okay, got back to my laptop and looked a bit closer at the problem and the issues described here and actually think it is just the use of picture-elements in general. I added `element` as an option for this exact reason, as watching at the top level of the card is just a really bad idea with this card.

So, I think the main problem is actually how this is being used with picture-elements cards. Instead of using this card at the top-level you should be using it at the element level per the docs: https://github.com/iantrich/config-template-card#picture-elements-card-example

This will be the best for performance and will not cause the full card to re-render.

I'm hopeful that changing your configs in such a way will fix your issues and unless anyone here has a good reason to do picture-elements at the card-level, I think I'll disallow `type: picture-elements` from being used in such a manner.

**Comment by bkr1969** [comment]
Can you modify the simple config example I posted to show this? I'm leery about all the sunlight opacities and whatnot in my actual config.

**Comment by iantrich** [comment]
```
type: picture-elements
image: https://demo.home-assistant.io/stub_config/floorplan.png
elements:
  - type: custom:config-template-card
    entities:
      - switch.ceiling_fan
    element:
      type: image
      tap_action:
        action: toggle
      entity: switch.ceiling_fan
      image: /local/floorplan/famfanoff.png
      state_image:
        'on': /local/floorplan/famfanon.png
      style:
        top: 80.3%
        left: 77.6%
```

**Comment by bkr1969** [comment]
Nope.  That does not work at all with the sunlight opacity or anything else as far as light-blend-mode.

**Comment by bkr1969** [comment]
That doesn't seem to work with the sunlight opacity.  Here's the start of my original code:

```
type: custom:config-template-card
entities:
  - sun.sun
  - light.couch_light
  - light.table_light
  - light.chair_light
  - light.main_light
  - light.dinning_room
  - light.kitchen
  - light.0420038868c63abfbb83
  - light.sink
  - switch.patio
  - switch.coach_lights
  - light.living_room_lamp
  - light.living_room
  - light.front_room
  - light.foyer
  - switch.warmer
  - light.stairs
  - switch.moms_lamp
  - light.entry_hall
  - switch.liv_tree
  - binary_sensor.shelly1_500291ed53d5_input
  - person.brian_2
  - person.desiree
  - person.mia
  - lock.front_door
  - sensor.front_door_battery
  - switch.flower_bed_4
  - switch.front_yard_2
  - switch.side_yard_2
  - switch.main_backyard_2
  - switch.patio_3
  - switch.north_backyard_2
  - input_boolean.downtherm
  - climate.downstairs
  - sensor.recycle
  - sensor.trash
  - alarm_control_panel.alarm_com
  - weather.home
  - vacuum.downstairsroomba
  - switch.ceiling_fan
  - input_boolean.mominfo
  - switch.moms_lamp
  - switch.office
  - switch.purple
  - switch.white_fan
  - input_boolean.momalexa
  - media_player.mom_s_office
  - input_boolean.famalexa
  - media_player.tv_room
card:
  type: picture-elements
  image: /local/floorplan/darkpanel.png
  style: |
    ha-card:first-child {
      background: rgba(42, 46, 48, 1)
    }
  elements:
    - type: image
      action: none
      entity: sun.sun
      state_image:
        above_horizon: /local/floorplan/lightpanel.png
        below_horizon: /local/floorplan/transp.png
      style:
        height: 100%
        left: 50%
        mix-blend-mode: lighten
        opacity: ${ states['sensor.sunlight_opacity'].state }
        top: 50%
        width: 100%
      tap_action:
        action: none
      hold_action:
        action: none
    - type: image
      action: none
      entity: light.couch_light
      hold_action:
        action: none
      image: /local/floorplan/transcouch.png
      style:
        left: 50%
        mix-blend-mode: lighten
        opacity: >-
          ${states['light.couch_light'].state === 'on' ?
          (states['light.couch_light'].attributes.brightness / 255) :'0'}
        top: 50%
        width: 100%
        height: 100%
      tap_action:
        action: none
```
How do I modify it? I'm getting errors that I can't completely read because they're off the screen to the left, but It looks like there's a problem with my daylight png and definitely something with the sunlight opacity.

**Comment by bkr1969** [comment]
Even with this completely cut down card, the daylight image doesn't line up and the opacity does not work.
```
type: picture-elements
image: /local/floorplan/darkpanel.png
style: |
  ha-card:first-child {
    background: rgba(42, 46, 48, 1)
  }
elements:
  - type: custom:config-template-card
    entities:
      - sun.sun
    element:
      type: image
      action: none
      entity: sun.sun
      state_image:
        above_horizon: /local/floorplan/lightpanel.png
        below_horizon: /local/floorplan/transp.png
      style:
        height: 100%
        left: 50%
        mix-blend-mode: lighten
        opacity: ${ states['sensor.sunlight_opacity'].state }
        top: 10%
        width: 100%
      tap_action:
        action: none
      hold_action:
        action: none
```

**Comment by bkr1969** [comment]
OK...I'm giving a lot of feedback to hopefully pin this down.  First of all, the simple example you reposted does not allow placement of the fan image.  It always hangs off the bottom left corner barely visible. When I take it out and put it as an element of the picture-elements card, it is moveable but always has a grey circle around it until I click somewhere else.  That isn't right.  I got my daylight image to align by making it an element of picture-element, but because it is now not nested inside the config-template card, the opacity doesn't work at all.  I'm beginning to feel like there is no work around for this.  Picture-elements has to be a child of config-template, not the other way around.

**Comment by iantrich** [comment]
I messed up the example. style should be at the level of config-template-card, not under element

**Comment by bkr1969** [comment]
Helpful, but I still cannot get my images for lights on to show, or if I get them to show they don'y toggletoggle.  Take a look at this and tell me what's wrong (there are multiple alerts not fully visible off the left side of the screen):
```
type: picture-elements
image: /local/floorplan/darkpanel.png
style: |
  ha-card:first-child {
    background: rgba(42, 46, 48, 1)
  }
elements:
  - type: custom:config-template-card
    entities:
      - sun.sun
      - light.couch_light
    elements:
      - type: image
        image: /local/floorplan/transp.png
        action: none
        state_image:
          above_horizon: /local/floorplan/lightpanel.png
          below_horizon: /local/floorplan/transp.png
        style:
          height: 100%
          left: 50%
          mix-blend-mode: lighten
          opacity: ${ states['sensor.sunlight_opacity'].state }
          top: 50%
          width: 100%
        tap_action:
          action: none
        hold_action:
          action: none
      - type: image
        image: /local/floorplan/transcouch.png
        action: none
        style:
          left: 50%
          mix-blend-mode: lighten
          opacity: >-
            ${states['light.couch_light'].state === 'on' ?
            (states['light.couch_light'].attributes.brightness / 255) :'0'}
          top: 50%
          width: 100%
          height: 100%
```

**Comment by smonesi** [comment]
I think there is something wrong about the `style` placement: if I put it at the same level as config-template-card I get a "No style defined for element" error (this happens even with the documented koala example), if I put it inside element I got no error but the style seems to be ignored.

**Comment by bkr1969** [comment]
Exactly what I’m finding.

**Comment by bkr1969** [comment]
So I've managed to get the lamp light image to show up and work, but the daylight opacity is not.  I think if we can crack that all will be good.

**Comment by iantrich** [comment]
I think I see the big related to style. For testing can you try putting your actual style at the config-template-card level and then just an empty style tag in the element. It looks like just my error handling is incorrect.

**Comment by bkr1969** [comment]
That's how I got the lamp image to show up, but why doesn't the opacity work?

**Comment by bkr1969** [comment]
OK...the opacity does work if stated explicitly (i.e. 50%) but will not pull from the brightness attribute for the light.  I'm guessing this could just be a simple formatting issue.  Thoughts?

**Comment by iantrich** [comment]
Okay, we'll it sounds like the initial issue is worked out and you can open another issue or go to the forums for your opacity issue. I'll make the change to fix the error handling and consider this fixed. Will also present a warning if used at the top-level of a picture-elements card.

**Comment by Koendejongh** [comment]
So can anyone provide me with a working example?

**Comment by iantrich** [comment]
https://github.com/iantrich/config-template-card/releases/tag/1.3.2

**Comment by iantrich** [comment]
> So can anyone provide me with a working example?

This should work with 1.3.2

```
type: picture-elements
image: https://demo.home-assistant.io/stub_config/floorplan.png
elements:
  - type: custom:config-template-card
    style:
      top: 80.3%
      left: 77.6%
    entities:
      - switch.ceiling_fan
    element:
      type: image
      tap_action:
        action: toggle
      entity: switch.ceiling_fan
      image: /local/floorplan/famfanoff.png
      state_image:
        'on': /local/floorplan/famfanon.png
```

**Comment by iantrich** [comment]
Looks like the release failed. That's what I get for trying to code from my phone. Will take a look when I get back to my computer.

**Comment by bkr1969** [comment]
I believe the opacity issue is with the con-fig-template card since it works with picture-elements.  Is that the new issue you'd like me to open?

**Comment by iantrich** [comment]
> I believe the opacity issue is with the config-template card since it works fine with picture-elements. Is that the new issue you'd like me to open?

Yes, it is not related to this initial issue

**Comment by bkr1969** [comment]
@ Koendejongh, here is the begining of mine (quite a chore to change everything over).  Basically, you have to have a new costom-config-card for each entity.  Remember, though, that the transparencies are not working yet.

```
type: picture-elements
image: /local/floorplan/darkpanel.png
style: |
  ha-card:first-child {
    background: rgba(42, 46, 48, 1)
  }
elements:
  - type: custom:config-template-card
    variables:
      - states['sensor.sunlight_opacity'].state
    entities:
      - sun.sun
      - sensor.sunlight_opacity
    element:
      type: image
      entity: sun.sun
      image: /local/floorplan/transp.png
      action: none
      state_image:
        above_horizon: /local/floorplan/lightpanel.png
        below_horizon: /local/floorplan/transp.png
      style:
        left: 50%
        mix-blend-mode: lighten
        opacity: '?{ states[''sensor.sunlight_opacity''].state }'
    style:
      height: 100%
      left: 50%
      mix-blend-mode: lighten
      opacity: 10%
      top: 50%
      width: 100%
    tap_action:
      action: none
    hold_action:
      action: none
  - type: custom:config-template-card
    variables:
      COUCHB: states['light.couch_light'].attributes.brightness
    entities:
      - light.couch_light
    element:
      type: image
      entity: light.couch_light
      image: /local/floorplan/transp.png
      state_image:
        'on': /local/floorplan/transcouch.png
      tap_action:
        action: none
      style:
        left: 50%
    style:
      mix-blend-mode: lighten
      opacity: '?{ states[''light.couch_light''].attributes.brightness / 255 }'
      width: 100%
      top: 50%
      left: 50%

**Comment by Nantero1** [comment]
@bkr1969 looking at your example: Is `?{ expression}` the same as `${ expression }`?

**Comment by iantrich** [comment]
It is not the same and likely why they are not working 😉

**Comment by bkr1969** [comment]
Those are all remnants of different trials.  The first is simply setting the opacity to the expression, the second tests for the condition, applies the expression if true, or the part after the colon if not.  The opacity didn't work with either method. (The ? was an error, but it was tested with $ as well...still didn't work)

**Comment by Nantero1** [comment]
I work usually with something like this for the opacity:
```yaml
        opacity: >-
          ${states['light.flurleuchte_2'].state === 'on' ?
          (states['light.flurleuchte_2'].attributes.brightness / 255) : '0'}
```

**Comment by Nantero1** [comment]
@iantrich 
> > So can anyone provide me with a working example?
> 
> This should work with 1.3.2
> 
> ```
> type: picture-elements
> image: https://demo.home-assistant.io/stub_config/floorplan.png
> elements:
>   - type: custom:config-template-card
>     style:
> ...

Tried... now only the component under `type: custom:config-template-card` is flickering. Of course, it's better than having everything flickering. But this is not a real "fix", the flickering is there and I can see the blue loading circle of a glimpse of a second.

Is it possible to overwrite the style somehow to disable the "blue loading circle" animation completely? I don't see any benefit in having a blue loading circle animation. Can one add a special "id" or style attribute to overwrite this behaviour? Any ideas?

I will try to Adblock the animation ^^, let's see if this helps.

**Comment by bkr1969** [comment]
> @ Koendejongh, here is the begining of mine (quite a chore to change everything over). Basically, you have to have a new costom-config-card for each entity. Remember, though, that the transparencies are not working yet.
> 
> ```
> type: picture-elements
> image: /local/floorplan/darkpanel.png
> style: |
>   ha-card:first-child {
>     background: rgba(42, 46, 48, 1)
>   }
> elements:
>   - type: custom:config-template-card
>     variables:
>       - states['sensor.sunlight_opacity'].state
>     entities:
>       - sun.sun
>       - sensor.sunlight_opacity
>     element:
>       type: image
>       entity: sun.sun
>       image: /local/floorplan/transp.png
>       action: none
>       state_image:
>         above_horizon: /local/floorplan/lightpanel.png
>         below_horizon: /local/floorplan/transp.png
>       style:
>         left: 50%
>         mix-blend-mode: lighten
>         opacity: '?{ states[''sensor.sunlight_opacity''].state }'
>     style:
>       height: 100%
>       left: 50%
>       mix-blend-mode: lighten
>       opacity: 10%
>       top: 50%
>       width: 100%
>     tap_action:
>       action: none
>     hold_action:
>       action: none
>   - type: custom:config-template-card
>     variables:
>       COUCHB: states['light.couch_light'].attributes.brightness
>     entities:
>       - light.couch_light
>     element:
>       type: image
>       entity: light.couch_light
>       image: /local/floorplan/transp.png
>       state_image:
>         'on': /local/floorplan/transcouch.png
>       tap_action:
>         action: none
>       style:
>         left: 50%
>     style:
>       mix-blend-mode: lighten
>       opacity: '?{ states[''light.couch_light''].attributes.brightness / 255 }'
>       width: 100%
>       top: 50%
>       left: 50%
> ```

**Comment by bkr1969** [comment]
It seems odd to me that my “…taking longer than 10 seconds…” messages are always only for one of my thermostats yet they both stop updating. Just restarted and got the message writhing 2 minutes

**Comment by vring0** [comment]
> @ Koendejongh, вот и начало моего (довольно утомительно, чтобы все изменить). По сути, у вас должна быть новая карта costom-config-card для каждой сущности. Однако помните, что прозрачные пленки еще не работают.
> 
> ```
> type: picture-elements
> image: /local/floorplan/darkpanel.png
> style: |
>   ha-card:first-child {
>     background: rgba(42, 46, 48, 1)
>   }
> elements:
>   - type: custom:config-template-card
>     variables:
>       - states['sensor.sunlight_opacity'].state
>     entities:
>       - sun.sun
>       - sensor.sunlight_opacity
>     element:
>       type: image
>       entity: sun.sun
>       image: /local/floorplan/transp.png
>       action: none
>       state_image:
>         above_horizon: /local/floorplan/lightpanel.png
>         below_horizon: /local/floorplan/transp.png
>       style:
>         left: 50%
>     style:
>       height: 100%
>       left: 50%
>       mix-blend-mode: lighten
>       opacity: '${ states[''sensor.sunlight_opacity''].state }'
>       top: 50%
>       width: 100%
>     tap_action:
>       action: none
>     hold_action:
>       action: none
>   - type: custom:config-template-card
>     entities:
>       - light.couch_light
>     element:
>       type: image
>       entity: light.couch_light
>       image: /local/floorplan/transp.png
>       state_image:
>         'on': /local/floorplan/transcouch.png
>       tap_action:
>         action: none
>       style:
>         left: 50%
>     style:
>       mix-blend-mode: lighten
>       opacity: >-
>         ${states['light.couch_light'].state === 'on' ?
>         (states['light.couch_light'].attributes.brightness / 255) :'0'}
>       width: 100%
>       top: 50%
>       left: 50%
> ```

Very uncomfortable. Is this a temporary solution or are we waiting for a fix?

**Comment by bkr1969** [comment]
This is how it needs to be coded now to avoid screen refreshing on state changes (and how it technically should've been done all along). The only thing not working is the opacity for elements as the config-template-card will not currently pull states of entities.  There has been a probable fix for this and we're just waiting for it to be implemented.

**Comment by hballard94** [comment]
Hello, I've reformatted my code but amnot having any luck getting my LED lights to change according to the color wheel.`views:
  - title: Floorplan
    path: floorplan
    panel: true
    icon: mdi:floor-plan
    type: panel
    subview: false
    badges: []
    cards:
      - type: custom:config-template-card
        card:
          type: picture-elements
          image: /local/ui/floorplan/off_day.png
          elements:
            - type: image
              entity: sun.sun
              tap_action: none
              hold_action: none
              state_image:
                above_horizon: /local/transparent.png
                below_horizon: /local/ui/floorplan/off_night.png
                unavailable: /local/ui/floorplan/off_night.png
              style:
                top: 50%
                left: 50%
                width: 100%
            - type: custom:config-template-card
              entity: light.bulb_2
              tap_action: none
              hold_action: none
              state_image:
                'on': /local/ui/floorplan/lucas_lamp_on.png
                'off': /local/transparent.png
                unavailable: /local/transparent.png
              style:
                filter: >-
                  ${ "hue-rotate(" + (states['light.bulb_2'].attributes.hs_color
                  ? states['light.bulb_2'].attributes.hs_color[0] : 0) + "deg)"}
                top: 50%
                left: 50%
                width: 100%
                mix-blend-mode: lighten`

**Comment by ildar170975** [comment]
1. Do not place one CTC card inside another CTC card as you did:
```
      - type: custom:config-template-card
        card:
          type: picture-elements
          image: ...
          elements:
            ...
            - type: custom:config-template-card
              entity: ...
```
2. The whole syntax is wrong:
```
            - type: custom:config-template-card
              entity: ...
              ...
              state_image:
                ...
              style:
                filter: ...
```
Here is a correct syntax:
```
            - type: custom:config-template-card
              entities:
                - ...
              element:
                type: ...
                ... rest of other code of this element
```
3. As a recommendation - do not place the whole `picture-elements` card inside CTC:
```
      - type: custom:config-template-card
        card:
          type: picture-elements
          image: ...
```
Use CTC for particular elements.
4. In many case there is no need to use CTC for elements! Card-mod may be easily used to dynamically change CSS styles:
instead of
```
              style:
                filter: >-
                  ${ ... some dependency of some entity ... }
```
use
```
              card_mod:
                style: |
                  :host {
                    {% if .... %}
                      --my-filter: .....;
                    {% else %}
                      --my-filter: .....;
                    {% endif %}
                  }
              style:
                filter: var(--my-filter)
```

**Comment by Anoxym** [comment]
Hi,

I have the same problem on my code, but I don't know how to move things to get it works, i've tested to put CTC inside picture-elements but still blinking when time update or light's state change :

`
- type: 'custom:config-template-card'
  entities:
    - switch.lamp10
    - switch.lamp9
    - switch.lamp3
    - switch.balcony1
    - switch.lamp8
    - switch.lamp2
    - switch.lamp4
    - switch.lamp5
    - light.living
    - sensor.timenew
    - sensor.date_string
    - 
  card:
    ###################   Picture Elements Card    #####################
    type: picture-elements
    image: /local/ui/floorplan/main_night.png
    style: |
      ha-card:first-child {
        background: rgba(42, 46, 48, 1)
      }
    elements:
      
        #########################    DAY BASE FLOORPLAN OVERLAY    #########################

      - action: none
        entity: sun.sun
        hold_action:
          action: none
        state_image:
          above_horizon: /local/ui/floorplan/main_day.png
          below_horizon: /local/transparent.png
        style:
          height: 100%
          left: 50%
          mix-blend-mode: lighten
          opacity: 0%
          top: 50%
          width: 100%
        tap_action:
          action: none
        type: image
        #########################    INDIVIDUAL LIGHTS    #########################

      #########################    LIVING    #########################
      
      - action: none
        entity: switch.lamp10
        hold_action:
          action: none
        image: /local/ui/floorplan/lights/lamp10.png
        style:
          left: 50%
          mix-blend-mode: lighten
          opacity: "${states['switch.lamp10'].state === 'on' ? (states['switch.lamp10'].attributes.brightness / 255) : '0'}"
          top: 50%
          width: 100%
        tap_action:
          action: none
        type: image
        
      - action: none
        entity: switch.lamp9
        hold_action:
          action: none
        image: /local/ui/floorplan/lights/lamp9.png
        style:
          left: 50%
          mix-blend-mode: lighten
          opacity: "${states['switch.lamp9'].state === 'on' ? (states['switch.lamp9'].attributes.brightness / 255) : '0'}"
          top: 50%
          width: 100%
        tap_action:
          action: none
        type: image
        
      - action: none
        entity: switch.lamp8
        hold_action:
          action: none
        image: /local/ui/floorplan/lights/lamp8.png
        style:
          left: 50%
          mix-blend-mode: lighten
          opacity: "${states['switch.lamp8'].state === 'on' ? (states['switch.lamp8'].attributes.brightness / 255) : '0'}"
          top: 50%
          width: 100%
        tap_action:
          action: none
        type: image
        
      - action: none
        entity: switch.lamp2
        hold_action:
          action: none
        image: /local/ui/floorplan/lights/lamp2.png
        style:
          left: 50%
          mix-blend-mode: lighten
          opacity: "${states['switch.lamp2'].state === 'on' ? (states['switch.lamp2'].attributes.brightness / 255) : '0'}"
          top: 50%
          width: 100%
        tap_action:
          action: none
        type: image

      - action: none
        entity: light.living
        hold_action:
          action: none
        image: /local/ui/floorplan/lights/wled-living.png
        style:
          filter: >-
            ${ "hue-rotate(" +
            (states['light.living'].attributes.hs_color  ?
            states['light.living'].attributes.hs_color[0] : 0) + "deg)"}
          left: 50%
          mix-blend-mode: lighten
          opacity: "${states['light.living'].state === 'on' ? (states['light.living'].attributes.brightness / 255) : '0'}"
          top: 50%
          width: 100%
        tap_action:
          action: none
        type: image
        
      #########################    OUTSIDE    #########################    
      
      - action: none
        entity: switch.lamp3
        hold_action:
          action: none
        image: /local/ui/floorplan/lights/lamp3.png
        style:
          left: 50%
          mix-blend-mode: lighten
          opacity: "${states['switch.lamp3'].state === 'on' ? (states['switch.lamp3'].attributes.brightness / 255) : '0'}"
          top: 50%
          width: 100%
        tap_action:
          action: none
        type: image
        
      - action: none
        entity: switch.balcony1
        hold_action:
          action: none
        image: /local/ui/floorplan/lights/balcony1.png
        style:
          left: 50%
          mix-blend-mode: lighten
          opacity: "${states['switch.balcony1'].state === 'on' ? (states['switch.balcony1'].attributes.brightness / 255) : '0'}"
          top: 50%
          width: 100%
        tap_action:
          action: none
        type: image
        
      #########################    ENTRANCE    #########################    
      
      - action: none
        entity: switch.lamp4
        hold_action:
          action: none
        image: /local/ui/floorplan/lights/4.png
        style:
          left: 50%
          mix-blend-mode: lighten
          opacity: "${states['switch.lamp4'].state === 'on' ? (states['switch.lamp4'].attributes.brightness / 255) : '0'}"
          top: 50%
          width: 100%
        tap_action:
          action: none
        type: image
        
      - action: none
        entity: switch.lamp5
        hold_action:
          action: none
        image: /local/ui/floorplan/lights/5.png
        style:
          left: 50%
          mix-blend-mode: lighten
          opacity: "${states['switch.lamp5'].state === 'on' ? (states['switch.lamp5'].attributes.brightness / 255) : '0'}"
          top: 50%
          width: 100%
        tap_action:
          action: none
        type: image
        
        #########################    CAMERAS POPUP on click    #########################

      - type: icon
        icon: mdi:cctv
        style:
          height: 24px
          left: 96.3%
          top: 50.5%
          width: 24px
        tap_action:
          action: call-service
          service: browser_mod.popup
          data:
            size: wide
            dismissable: true
            autoclose: false
            content:
              camera_view: live
              type: picture-glance
              entities: []
              camera_image: camera.door1_sub
              
      - type: icon
        icon: mdi:doorbell-video
        style:
          height: 24px
          left: 87%
          top: 94%
          width: 24px
        tap_action:
          action: call-service
          service: browser_mod.popup
          data:
            size: wide
            dismissable: true
            autoclose: false
            content:
              camera_view: live
              type: picture-glance
              entities: []
              camera_image: camera.doorbell_fluent
              
      - type: icon
        icon: mdi:cctv
        style:
          height: 24px
          left: 92%
          top: 16%
          width: 24px
        tap_action:
          action: call-service
          service: browser_mod.popup
          data:
            size: wide
            dismissable: true
            autoclose: false
            content:
              camera_view: live
              type: picture-glance
              entities: []
              camera_image: camera.door2_sub
              
      - type: icon
        icon: mdi:cctv
        style:
          height: 24px
          left: 43%
          top: 16%
          width: 24px
        tap_action:
          action: call-service
          service: browser_mod.popup
          data:
            size: wide
            dismissable: true
            autoclose: false
            content:
              camera_view: live
              type: picture-glance
              entities: []
              camera_image: camera.garage_sub


        #########################    INDIVIDUAL LIGHT ICONS    #########################

      - entity: switch.lamp10
        type: custom:button-card
        style:
          left: 51.5%
          top: 69%
        template:
          - light

      - entity: switch.lamp9
        type: custom:button-card
        style:
          left: 44.5%
          top: 40.5%
        template:
          - light

      - entity: switch.lamp8
        type: custom:button-card
        style:
          left: 50.5%
          top: 40%
        template:
          - light
         
      - entity: light.living
        type: custom:button-card
        style:
          left: 49.7%
          top: 49.9%
        template:
          - light

      - entity: switch.lamp2
        type: custom:button-card
        style:
          left: 48.3%
          top: 61.5%
        template:
          - light

      - entity: switch.lamp3
        type: custom:button-card
        style:
          left: 83%
          top: 59%
        template:
          - light

      - entity: switch.balcony1
        type: custom:button-card
        style:
          left: 52.5%
          top: 79.7%
        template:
          - light
          
      - entity: switch.lamp4
        type: custom:button-card
        style:
          left: 64%
          top: 44%
        template:
          - light

      - entity: switch.lamp5
        type: custom:button-card
        style:
          left: 65%
          top: 49.5%
        template:
          - light
        
      - action: none
        hold_action:
          action: none
        image: /local/ui/sidebar/sidebarBG2.png
        style:
          height: 100%
          left: 11.73828125%
          top: 50%
          width: 23.4765625%
        tap_action:
          action: none
        type: image
        
        #################################################################
        #                                                               #
        #                             SIDEBAR                           #
        #                                                               #
        #################################################################

        #########################    DATETIME    #########################
      - entity: sensor.timenew
        hold_action:
          action: none
        style:
          color: 'rgba(255, 255, 255, 0.7)'
          font-size: 4.20vw
          font-weight: 200
          left: 2.7%
          letter-spacing: '-0.05vw'
          max-width: 1px
          top: 10%
        tap_action:
          action: none
        type: state-label
      - entity: sensor.date_string
        hold_action:
          action: none
        style:
          color: 'rgba(255, 255, 255, 0.3)'
          font-size: 1.3vw
          font-weight: 300
          left: 18.3%
          letter-spacing: '-0.05vw'
          text-align: left
          top: 17%
          width: 30%
        tap_action:
          action: none
        type: state-label`

Can someone help me to modify this code?
Thanks!

**Comment by ildar170975** [comment]
@Anoxym 
Using CTC for the whole picture-elements is not recommended.
CTC should be used to [style a single element](https://github.com/iantrich/config-template-card#picture-elements-card-example).

**Comment by Christian-CFP** [comment]
I am using the floorplan from [lukevink](https://github.com/lukevink/hass-config-lajv/commits?author=lukevink).
First I am a HA beginner but I did get it to run (Cleaning and Room Control) but I have the flickering issue. I read a lot and I get only (for me) some kind of sh** CTC inside outside picture element inside outside.

Can somebody explain me in easy what I have to do to get ridd of this "reloading" site if light is changing. THX! 

The light images I changed it from - action:none to - type: custom:config-template-card (they work) but there are so many - action: none types and I do not know where to start to check?
